daemon/daemon.go

```diff
@@ -1149,7 +1150,12 @@ func (daemon *Daemon) networkOptions(dconfig *config.Config, pg plugingetter.Plu
 		options = append(options, nwconfig.OptionPluginGetter(pg))
 	}
 
-	options = append(options, nwconfig.OptionNetworkControlPlaneMTU(dconfig.NetworkControlPlaneMTU))
+	if dconfig.NetworkControlPlaneMTU != 0 {
+		if dconfig.NetworkControlPlaneMTU < 500 {
+			return nil, errors.Errorf("Network Control plane MTU (%d) out of range. Minimum value is 500", dconfig.NetworkControlPlaneMTU)
+		}
+		options = append(options, nwconfig.OptionNetworkControlPlaneMTU(dconfig.NetworkControlPlaneMTU))
```

Comment:
libnetwork now also prints a warning; https://github.com/docker/libnetwork/pull/1892, but haven't checked yet how that looks when using